app/api/auth.py

```diff
@@ -94,6 +94,10 @@ def login_user(provider, auth_code):
             'client_secret': provider_class.get_client_secret(),
             'code': auth_code
         }
+        if not payload['client_id']:
+            return 'Client_id not configured on server', 500
+        if not payload['client_secret']:
+            return 'Client_secret no configured on server', 500
```

Comment:
500 is internal server error. You should never send 500 to clients, it is an unexpected error

Secondly, just OR the condition and send a proper JSON telling which social auth is not enabled

JSON should be consistent with JSONAPI spec and in line with other errors returned from the server

Reply:
@iamareebjamal Yeah got it. Will remove the 500 error and just throw an error stating that the facebook login hasn't been configured.